src/validation.cpp

```diff
@@ -5594,8 +5592,7 @@ ChainstateManager::~ChainstateManager()
 
     m_versionbitscache.Clear();
 
-    // TODO: The warning cache should probably become non-global
-    for (auto& i : warningcache) {
+    for (auto& i : m_warningcache) {
```

Comment:
This loop can be dropped entirely, no? The destructor's already running, so `m_warningcache` will be deleted momentarily too.

Reply:
Makes sense, didn't want to change much since its my first PR. Removed it with 8bba010